lib/pages/home_page.dart

```diff
@@ -13,54 +14,131 @@ class HomePage extends StatefulWidget {
+          );
+        }
+        if (snapshot.connectionState == ConnectionState.done) {
+          if (gps == LocationPermission.denied ||
+              gps == LocationPermission.unableToDetermine) {
+            return Scaffold(
+                appBar: AppBar(
+                  title: const Text('Home'),
+                ),
+                body: SizedBox(
+                    height: double.infinity,
+                    width: double.infinity,
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.center,
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        const Text("GPS PERMISSION REQUIRED TO SHOW MAP"),
+                        const Text(
+                            "This app requires permission to access GPS as the core elements cannot function without it",
+                            textAlign: TextAlign.center),
+                        const ElevatedButton(
+                            onPressed: Geolocator.openAppSettings,
+                            child: Text("Open app settings")),
+                        const SizedBox(),
+                        ElevatedButton(
+                            onPressed: refreshBtn, child: const Text("Refresh"))
+                      ],
+                    )));
+          } else if (gps == LocationPermission.deniedForever) {
+            return Scaffold(
+                appBar: AppBar(
+                  title: const Text('Home'),
+                ),
+                body: SizedBox(
+                    height: double.infinity,
+                    width: double.infinity,
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.center,
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        const Text("GPS PERMISSION DENIED FOREVER"),
+                        const Text(
+                            "This app requires permission to access GPS as the core elements cannot function without it",
+                            textAlign: TextAlign.center),
+                        const ElevatedButton(
+                            onPressed: Geolocator.openAppSettings,
+                            child: Text(
+                              "Open app settings",
+                            )),
+                        const SizedBox(),
+                        ElevatedButton(
+                            onPressed: refreshBtn, child: const Text("Refresh"))
+                      ],
+                    )));
+          }
+          return Scaffold(
+            appBar: AppBar(
+              title: const Text('Home'),
+              
+            ),
+            //TODO: bottomNavigationBar: BottomAppBar(), 
+            body: SizedBox(
+              height: double.infinity,
+              width: double.infinity,
+              child: Scaffold(
```

Comment:
Máš tu 2x vnořený `Scaffold`, zítra ti řeknu proč to může být problém a co všechno Scaffold krom toho že to je Widget dělá.